am-configure: enhance SS user configuration

mamedin opened the pull request:
This commit will:
    
    - Enhance configure SS user #1: Detect and delete the default SS test when id=1
    on SS database and never was used. This way the role avoids to delete a
    possible "test" user added by an admin or by this role.
    
    - Enhance configure SS user #2: Check whether SS user already exists. In this
    case the SS user is not modified.
    
    - Enhance configure Dashboard user: Fix conditional typo at "Create Dashboard
    user and register pipeline on SS"
    
    - Add sqlite3 as OS dependency (Needed to check SS user on SS database)

Fixes https://github.com/artefactual-labs/ansible-archivematica-src/issues/222

mamedin committed:
am-configure: enhance SS user configuration

This commit will:

- Enhance configure SS user #1: Detect and delete the default SS test when id=1
on SS database and never was used. This way the role avoids to delete a
possible "test" user added by an admin or by this role.

- Enhance configure SS user #2: Check whether SS user already exists. In this
case the SS user is not modified.

- Enhance configure Dashboard user: Fix conditional typo at "Create Dashboard
user and register pipeline on SS"

- Add sqlite3 as OS dependency (Needed to check SS user on SS database)

scollazo commented on tasks/configure.yml:
```diff
@@ -5,16 +5,15 @@
-    executable: /bin/bash
-  environment: "{{ archivematica_src_ss_environment }}"
-  when: "archivematica_src_configure_ss|bool"
+  command: sqlite3  "{{archivematica_src_ss_environment['SS_DB_NAME']}}" "select username from auth_user where username='{{ archivematica_src_configure_ss_user }}';"
```
Does this mean that SS 0.11 won't get configured? (The "test" user is created as part of migrations)

scollazo reviewed (changes requested):
You should be using a more meaningful commit message. What about "configure: improve ss user handling". Anything that is more or less descriptive about wht are you doing it's better than "fix some issues".

mamedin commented:
I changed commit message and I enabled again the task to delete default SS test user, but adding a new check to ensure this is the user added on migrations. This way a "test" user added by this role or sysadmin will not be deleted.

Finally I added sqlite3 as OS dependency.

It is ready for CR again.
